status-loop/queryValidators.js

```diff
@@ -214,35 +220,80 @@ async function queryValidators() {
 	await Promise.all(
 		campaigns.map(c =>
 			getStatusOfCampaign(c).then(async status => {
-				const [fundsDistributedRatio, usdEstimate] = await Promise.all([
-					getDistributedFunds(c, status.lastApprovedBalances),
-					getEstimateInUsd(c),
-				])
-				const statusObj = {
-					...c.status,
-					...status,
-					lastChecked: Date.now(),
-					usdEstimate,
-				}
-				// If the status was closed we don't want to update the funds distribution ratio as it will be 100%
-				if (status.humanFriendlyName !== 'Closed') {
-					statusObj.fundsDistributedRatio = fundsDistributedRatio
+				// skipping campaign if a validator is failing
+				if (status.error === 404) {
+					console.log(`Skipping campaign ${c.id} with status ${c.status.name}`)
+					return Promise.resolve()
 				}
+				return handleNewCampaignStatus(c, status, campaignsCol)
+			})
+		)
+	)
+}
 
-				if (status.humanFriendlyName === 'Completed' && !statusObj.closedDate) {
-					statusObj.closedDate = Date.now()
-				}
+async function handleValidatorCalls(leaderUrl, followerUrl, newStateUrl) {
+	let failed_calls = false
+	let dataLeader
+	let dataFollower
+	let dataLatestNewState
+	try {
```

Comment:
Can't you use `await Promise.all([...])` inside try catch or `Promise.all([ ... ].map(pr => pr.catch(...)))`. You can detect the the failed request but they will be send in parallel.